pypeit/coadd2d.py

```diff
@@ -550,6 +562,8 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None, basename=None):
 
         show = self.show if show is None else show
         show_peaks = self.show_peaks if show_peaks is None else show_peaks
+        # TODO I think this is incorrect. I don't know how to instanatiate this class from an existing mask
+        # because everthying is hidden behind exceedingly complicated bitmasks.
```

Comment:
The API for PypeIt image is impossible to understand because all the arguments are optional. Furthermore, sometimes we need to instantiate a PypeIt image from existing images without running the PypeIt image processing. In those situations (i.e. JWST, here in coadd2d, but also in quicklook), one has a mask for the image already. This is the combination of a bad pixel mask, CR mask, and possibly other masking. In this situation the mask does not adhere to our bit systems which is anyway. In those instances I need to know how to instantiate a PypeIt image with an existing mask. Improvements would entail making the API have required arguments, since this everything is optional is a very poor model. And also possibly more clear information documentation on how to input a mask. You might think that the crmask is the thing to use here but I already went that route and it is incorrect since the CRmask is constantly updated when you run LAcosmic in the base classes, whereas I want a mask that is basically always there. Can you please explain how to instantiate the fullmask then from an existing boolean good pixel mask? I suspect this is wrong in multiple places.